plugin/pulpcore/plugin/download/asyncio/file.py

```diff
@@ -0,0 +1,46 @@
+import os
+from urllib.parse import urlparse
+
+from .base import BaseDownloader, DownloadResult
+
+
+class FileDownloader(BaseDownloader):
+    """
+    A downloader for reading files from the filesystem.
+
+    It provides digest and size validation along with computation of the digests needed to save the
+    file as an Artifact.
+    """
+
+    def __init__(self, uri, **kwargs):
+        """
+        Download files from a uri that starts with `file://`
+
+        Args:
+            uri (str): The uri to the file. This is expected to begin with `file://`
+            kwargs (dict): This accepts the parameters of
+                :class:`~pulpcore.plugin.download.asyncio.BaseDownloader`.
+        """
+        self.uri = uri
+        kwargs['custom_file_object'] = open(os.devnull, 'w')
+        p = urlparse.urlparse(uri)
+        self._path = os.path.abspath(os.path.join(p.netloc, p.path))
+        super().__init__(**kwargs)
+
+    async def run(self):
+        """
+        Read, validate, and compute digests on the `uri`.
+
+        This method provides the same return object type and documented in
+        :meth:`~pulpcore.plugin.download.asyncio.BaseDownloader.run`.
+        """
+        with open(self._path, 'r') as f_handle:
+            while True:
+                chunk = await f_handle.read(1024)
+                if not chunk:
+                    self._validate_size()
+                    self._validate_digests()
+                    break  # the reading is done
```

Comment:
I think you meant: 
```
self._validate_size()
self._validate_digests()
```
to be:
```
self.validate_size()
self.validate_digests()
```

Reply:
Yup, this was a leftover. Thank you! fixed in next push.